python/alibiexplainer/alibiexplainer/explainer.py

```diff
@@ -1,48 +1,59 @@
-import kfserving
+import json
+import logging
 from enum import Enum
-from typing import List, Any, Dict, Mapping, Optional
-import numpy as np
+from typing import List, Any, Mapping, Union
+
+import kfserving
 import kfserving.protocols.seldon_http as seldon
-from kfserving.protocols.seldon_http import SeldonRequestHandler
+import numpy as np
 import requests
-import json
-import logging
+from alibiexplainer.anchor_images import AnchorImages
 from alibiexplainer.anchor_tabular import AnchorTabular
-from kfserving.server import Protocol
+from alibiexplainer.anchor_text import AnchorText
+from kfserving.protocols.seldon_http import SeldonRequestHandler
 from kfserving.protocols.util import NumpyEncoder
-
+from kfserving.server import Protocol
 
 logging.basicConfig(level=kfserving.server.KFSERVER_LOGLEVEL)
 
 
 class ExplainerMethod(Enum):
     anchor_tabular = "anchor_tabular"
+    anchor_images = "anchor_images"
+    anchor_text = "anchor_text"
 
     def __str__(self):
         return self.value
 
+
 class AlibiExplainer(kfserving.KFModel):
     def __init__(self,
                  name: str,
                  predict_url: str,
                  protocol: Protocol,
                  method: ExplainerMethod,
                  config: Mapping,
-                 explainer: object = None):
+                 explainer: object = None,
+                 host_header: str = None):
         super().__init__(name)
         self.predict_url = predict_url
+        self.host_header = host_header
         self.protocol = protocol
         self.method = method
 
         if self.method is ExplainerMethod.anchor_tabular:
-            self.wrapper = AnchorTabular(self._predict_fn,explainer,**config)
+            self.wrapper = AnchorTabular(self._predict_fn, explainer, **config)
+        elif self.method is ExplainerMethod.anchor_images:
+            self.wrapper = AnchorImages(self._predict_fn, explainer, **config)
+        elif self.method is ExplainerMethod.anchor_text:
+            self.wrapper = AnchorText(self._predict_fn, explainer, **config)
         else:
             raise NotImplementedError
 
     def load(self):
         pass
 
-    def _predict_fn(self, arr: np.ndarray) -> np.ndarray:
+    def _predict_fn(self, arr: Union[np.ndarray, List]) -> np.ndarray:
         if self.protocol == Protocol.seldon_http:
```

Comment:
looks this code for calling predictor can be shared between transformer and explainer. Will explainer ever need to call transformer ?

Reply:
If there is a pre-transformer I think the explainer needs to see the whole prediction graph as a black box so yes. It will receive the same input a user would send to the predict endpoint. Not so sure about the post-transformer as the explainer needs to get a class prediction back.